base/deprecated.jl

```diff
@@ -1632,6 +1632,15 @@ function SymTridiagonal(dv::AbstractVector{T}, ev::AbstractVector{S}) where {T,S
     SymTridiagonal(convert(Vector{R}, dv), convert(Vector{R}, ev))
 end
 
+# deprecate expm in favor of exp
+@deprecate expm!(A::StridedMatrix{<:LinAlg.BlasFloat}) exp!(A)
+@deprecate expm(A::StridedMatrix{<:LinAlg.BlasFloat}) exp(A)
+@deprecate expm(A::StridedMatrix{<:Integer}) exp(A)
+@deprecate expm(A::Hermitian) exp(A)
+@deprecate expm(A::Symmetric) exp(A)
+@deprecate expm(D::Diagonal) exp(D)
+@deprecate expm(x::Number) exp(x)
```

Comment:
Those functions simply won't exist in `Base` in the future so external packages would need a deprecation warning for a function deprecation. It's up to them if they want to define and export `expm` themselves, or switch to `Base.exp`.